examples/projects/dagster-slurm-example/dagster_slurm_example/defs/shell/shell_integration.py

```diff
@@ -67,3 +95,127 @@ def no_empty_order_check(
+        context_injector=injector,
+        message_reader=reader,
+    ) as session:
+        # 1) Write DAGSTER_PIPES_* bootstrap env to a file on the remote host
+        env = session.get_bootstrap_env_vars()
+        exports = "\n".join(f"export {k}={shlex.quote(v)}" for k, v in env.items())
+        ssh_check(
+            f"cat > {shlex.quote(remote_run_dir)}/pipes.env <<'EOF'\n{exports}\nEOF"
+        )
+
+        # 2) Build remote job script: activate env, source Pipes env, exec payload
+        bootstrap = textwrap.dedent(
+            f"""\
+            set -euo pipefail
+            source {shlex.quote(ACTIVATE_SH)}
+            source {shlex.quote(remote_run_dir)}/pipes.env
+            export JOB_OUTPUT_DIR={shlex.quote(remote_results)}
+            mkdir -p -- "$JOB_OUTPUT_DIR"
+            exec {shlex.quote(REMOTE_PY)} {shlex.quote(remote_run_dir)}/external_file.py
+            """
+        )
+        ssh_check(
+            "cat > {path}/job.sbatch <<'SB'\n#!/bin/bash\n{body}\nSB\nchmod +x {path}/job.sbatch".format(
+                path=shlex.quote(remote_run_dir),
+                body=bootstrap,
+            )
+        )
+
+        # 3) Submit via sbatch
+        opts = [
+            "-J",
+            "pipes_ext",
+            "-D",
+            remote_run_dir,
+            "-o",
+            f"{remote_run_dir}/slurm-%j.out",
+            "-e",
+            f"{remote_run_dir}/slurm-%j.err",
+            "-t",
+            TIME_LIMIT,
+            "-c",
+            CPUS,
+            f"--mem={MEM}",
+            # f"--mem-per-cpu={MEM_PER_CPU}" #--mem and --mem-per-cpu cannot be together
+        ]
+        if PARTITION:
+            opts += ["-p", PARTITION]
+
+        submit_cmd = (
+            "sbatch "
+            + " ".join(shlex.quote(x) for x in opts)
+            + f" {shlex.quote(remote_run_dir)}/job.sbatch"
+        )
+        out = ssh_check(submit_cmd)
+
+        m = re.search(r"Submitted batch job (\d+)", out)
+        if not m:
+            raise RuntimeError(f"Could not parse job id from sbatch.\nstdout:\n{out}")
+        job_id = int(m.group(1))
+        context.log.info(f"Submitted job {job_id}")
+
+        # 4) Stream messages while polling job state
+        while True:
+            st = ssh_job_state(job_id)
+            if st in TERMINAL_STATES:
+                break
+            time.sleep(1.0)
+
+    yield Output(
+        {"job_id": job_id, "remote_run_dir": remote_run_dir},
+        metadata={"job_id": job_id, "remote_run_dir": remote_run_dir},
+    )
+
+
+slurm_submit_pipes_ssh2 = make_slurm_pipes_asset(
+    name="slurm_submit_pipes_vanilla",
+    asset_key=["dse", "orders_with_pipes"],
+    local_payload="../../../../dagster-slurm-example-hpc-workload/dagster_slurm_example_hpc_workload/shell/shell_external.py",
+    extras={"foo": "bar"},
+    extra_env={
+        "MY_ENV_VAR_IN_SUBPROCESS": "hello-from-dagster",
+    },
+)
+
+slurm_submit_pipes_ray = make_slurm_pipes_asset(
```

Comment:
fails with


```
Local stderr (on error):
date: invalid argument 's' for -I
date: invalid argument 's' for -I

subprocess.CalledProcessError: Command 'bash /tmp/dagster_local_runs/pipes/b4747aad-e4de-47f3-a034-f70fa9752fb3/job.sh' returned non-zero exit status 126.
```

otherwise similar to the 1st error observation